### ISSUE TITLE
panic (Invalid memory adress) on swarm update

### ISSUE THREAD

**Issue by ZeissS** [giantswarm/cli#407]
I removed a dependency in a components with ~10 dependencies. All internal dependencies, nothing across services.

```
$ swarm version
swarm version 0.20.0+git, build 577c7c2ba2

$ swarm update simple-deps.json
Updating service 'simple-deps' using definition in 'simple-deps.json'.
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x0 pc=0x862cf]

goroutine 1 [running]:
github.com/giantswarm/user-config.diffComponentImage(0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/usr/code/.gobuild/src/github.com/giantswarm/user-config/diff.go:239 +0x29f
github.com/giantswarm/user-config.ComponentDiff(0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/usr/code/.gobuild/src/github.com/giantswarm/user-config/diff.go:220 +0xc4
github.com/giantswarm/user-config.diffComponentUpdated(0x208a2b2c0, 0x208a88600, 0x0, 0x0, 0x0)
	/usr/code/.gobuild/src/github.com/giantswarm/user-config/diff.go:184 +0x236
github.com/giantswarm/user-config.ServiceDiff(0x20890aec0, 0xb, 0x208a2b2c0, 0x2089e4530, 0xb, 0x208a88600, 0x0, 0x0, 0x0)
	/usr/code/.gobuild/src/github.com/giantswarm/user-config/diff.go:122 +0x2fd
main.showUpdateDiffPlan(0x7fff5fbffc9b, 0x10, 0x208b8aa00, 0x9ec, 0x208b8b400, 0x93a, 0x0, 0x0)
	/usr/code/update.go:202 +0x1ef
main.updateService(0x7fff5fbffc9b, 0x10, 0x0, 0x0)
	/usr/code/update.go:150 +0x7b4
main.updateCmdRun(0x6ecbc0, 0x20883a450, 0x1, 0x3)
	/usr/code/update.go:100 +0x250
github.com/spf13/cobra.(*Command).execute(0x6ecbc0, 0x208851740, 0x3, 0x4, 0x0, 0x0)
	/usr/code/.gobuild/src/github.com/spf13/cobra/command.go:530 +0x82f
github.com/spf13/cobra.(*Command).Execute(0x6eb6c0, 0x0, 0x0)
	/usr/code/.gobuild/src/github.com/spf13/cobra/command.go:613 +0x414
main.main()
	/usr/code/main.go:97 +0xc62

goroutine 5 [syscall]:
os/signal.loop()
	/usr/src/go/src/os/signal/signal_unix.go:21 +0x1f
created by os/signal.init·1
	/usr/src/go/src/os/signal/signal_unix.go:27 +0x35

goroutine 14 [IO wait]:
net.(*pollDesc).Wait(0x2087a0220, 0x72, 0x0, 0x0)
	/usr/src/go/src/net/fd_poll_runtime.go:84 +0x47
net.(*pollDesc).WaitRead(0x2087a0220, 0x0, 0x0)
	/usr/src/go/src/net/fd_poll_runtime.go:89 +0x43
net.(*netFD).Read(0x2087a01c0, 0x208859000, 0x1000, 0x1000, 0x0, 0x22087a3e70, 0x208b29eb0)
	/usr/src/go/src/net/fd_unix.go:242 +0x40f
net.(*conn).Read(0x2087d2090, 0x208859000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/src/go/src/net/net.go:121 +0xdc
crypto/tls.(*block).readFromUntil(0x2088624b0, 0x22087a6e88, 0x2087d2090, 0x5, 0x0, 0x0)
	/usr/src/go/src/crypto/tls/conn.go:454 +0xe6
crypto/tls.(*Conn).readRecord(0x20889e2c0, 0x17, 0x0, 0x0)
	/usr/src/go/src/crypto/tls/conn.go:539 +0x2da
crypto/tls.(*Conn).Read(0x20889e2c0, 0x208858000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/src/go/src/crypto/tls/conn.go:904 +0x166
net/http.noteEOFReader.Read(0x22087abbc8, 0x20889e2c0, 0x2088a8058, 0x208858000, 0x1000, 0x1000, 0x3c2040, 0x0, 0x0)
	/usr/src/go/src/net/http/transport.go:1270 +0x6e
net/http.(*noteEOFReader).Read(0x208936c60, 0x208858000, 0x1000, 0x1000, 0x2087a2000, 0x0, 0x0)
	<autogenerated>:125 +0xd4
bufio.(*Reader).fill(0x2089ef6e0)
	/usr/src/go/src/bufio/bufio.go:97 +0x1ce
bufio.(*Reader).Peek(0x2089ef6e0, 0x1, 0x0, 0x0, 0x0, 0x0, 0x0)
	/usr/src/go/src/bufio/bufio.go:132 +0xf0
net/http.(*persistConn).readLoop(0x2088a8000)
	/usr/src/go/src/net/http/transport.go:842 +0xa4
created by net/http.(*Transport).dialConn
	/usr/src/go/src/net/http/transport.go:660 +0xc9f

goroutine 15 [select]:
net/http.(*persistConn).writeLoop(0x2088a8000)
	/usr/src/go/src/net/http/transport.go:945 +0x41d
created by net/http.(*Transport).dialConn
	/usr/src/go/src/net/http/transport.go:661 +0xcbc

goroutine 21 [runnable]:
net/http.(*persistConn).readLoop(0x208b1b080)
	/usr/src/go/src/net/http/transport.go:928 +0x9ce
created by net/http.(*Transport).dialConn
	/usr/src/go/src/net/http/transport.go:660 +0xc9f

goroutine 11 [sleep]:
net.func·019()
	/usr/src/go/src/net/dnsclient_unix.go:240 +0x5a
created by net.loadConfig
	/usr/src/go/src/net/dnsclient_unix.go:269 +0x20c

goroutine 22 [select]:
net/http.(*persistConn).writeLoop(0x208b1b080)
	/usr/src/go/src/net/http/transport.go:945 +0x41d
created by net/http.(*Transport).dialConn
	/usr/src/go/src/net/http/transport.go:661 +0xcbc
```

**Comment by ZeissS** [comment]
FYI @ewoutp 

I will try to find the cause of this, ok?

**Comment by ZeissS** [comment]
```
{
  "name": "simple-deps",
  "components": {
    "deps": {
      "expose": [
        {
          "port": "8005",
          "component": "deps/dep1",
          "target_port": "80/tcp"
        },
        {
          "port": "8006",
          "component": "deps/dep2",
          "target_port": "80/tcp"
        }
      ]
    },
    "deps/dep1": {
      "image": "zeisss/static-website",
      "ports": [
        "80/tcp"
      ]
    },
    "deps/dep2": {
      "image": "zeisss/static-website",
      "ports": [
        "80/tcp"
      ]
    },
    "simple": {
      "expose": [
        {
          "port": "8001",
          "component": "simple/web",
          "target_port": "80/tcp"
        },
        {
          "port": "8002",
          "component": "simple/web",
          "target_port": "81/tcp"
        },
        {
          "port": "8003",
          "component": "simple/web",
          "target_port": 8080
        },
        {
          "port": "8004",
          "component": "simple/dep3",
          "target_port": "80/tcp"
        }
      ]
    },
    "simple/dep3": {
      "image": "zeisss/static-website",
      "ports": [
        "80/tcp"
      ],
      "domains": {
        "80/tcp": [
          "simple-deps.zeisss.leaseweb-neofonie.giantswarm.io"
        ]
      }
    },
    "simple/web": {
      "image": "alpine",
      "ports": [
        "80/tcp",
        "81/tcp",
        8080
      ],
      "args": [
        "sleep",
        "5000000"
      ],
      "links": [
        {
          "component": "simple",
          "alias": "othername",
          "target_port": "8004/tcp"
        },
        {
          "component": "simple",
          "alias": "othername1",
          "target_port": "8004/tcp"
        },
        {
          "component": "simple",
          "alias": "othername2",
          "target_port": "8004/tcp"
        },
        {
          "component": "simple",
          "alias": "othername3",
          "target_port": "8004/tcp"
        },
        {
          "component": "simple",
          "alias": "othername4",
          "target_port": "8004/tcp"
        },
        {
          "component": "simple",
          "alias": "othername6",
          "target_port": "8004/tcp"
        },
        {
          "component": "deps",
          "alias": "dependency",
          "target_port": "8005/tcp"
        }
      ]
    }
  }
}
```

**Comment by ewoutp** [comment]
Don't want to close this yet.
We should first use the fixed user-config in cli, api & app-service.

**Comment by ZeissS** [comment]
ACK. Github was a bit overactive ;)

**Comment by ZeissS** [comment]
This should be fixed by #406 which included user-config 0.28.1 with the fix.
